### DISCUSSION TITLE
Implementing boid flocking with OpenCilk

### DISCUSSION THREAD

**Discussion by behoppe** [OpenCilk/opencilk-project#249] [General]
Hi OpenCilkers. Several of us heard @libbyshoop present "boid flocking" as a [Peachy Assignment at EduPar](https://tcpp.cs.gsu.edu/curriculum/?q=system/files/peachy-edupar24.pdf) and thought it would be great to implement the HW in OpenCilk. (Libby's code is [here](https://github.com/csinparallel/PDCAssignments/tree/main/Boids).)  Has anyone made progress on that yet (e.g. @cleiserson @timkaler @neboat @chenxuhao @kyle-singer )?  Please let me/us know with a reply here, because I know that @libbyshoop is also interested in implementing boids with OpenCilk, and it could be helpful for boid-implementers to coordinate. Thank you!

**Comment by cleiserson** [comment]
Bruce,

I looked over Libby's code.  I don't see an OpenMP implementation, just the
C++ one.  I'd be happy to work with anyone who wanted to take on
parallelizing it.

Cheers,
Charles

—
Charles E. Leiserson
Professor of Computer Science and Engineering
Edwin Sibley Webster Professor in Electrical Engineering and Computer
Science
Margaret MacVicar Faculty Fellow
Member of the National Academy of Engineering <http://www.nae.edu>
Fellow of AAAS <http://www.aaas.org>, ACM <http://www.acm.org>, IEEE
<http://www.ieee.org>, and SIAM <http://www.siam.org>.



Computer Science and Artificial Intelligence Laboratory
<http://www.csail.mit.edu>
Massachusetts Institute of Technology <http://www.mit.edu>
32 Vassar Street, #32-G768
Cambridge, MA 02139
USA

MIT Phone: +1-617-253-5833
Web: http://people.csail.mit.edu/cel/
Email: ***@***.***
Administrative Assistant: Ms. Cree Bruins ***@***.***>
—

**Reply by behoppe** [reply]
Thanks @cleiserson. Did you see `#include "omp.h"`, `omp_set_num_threads(params.threads);`, and `omp_get_wtime();` in [Boids/OMPpsPlot/boids.c](https://github.com/csinparallel/PDCAssignments/blob/883c59b2298e3068c164ba161c4f586f15c4ddd8/Boids/OMPpsPlot/boids.c)?
